src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let obj = { ...state };
+
+  for (const properties of actions) {
+    const { extraData, keysToRemove, type } = properties;
+
+    switch (type) {
+      case 'addProperties':
+        for (const value in extraData) {
+          obj[value] = extraData[value];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const value of keysToRemove) {
+          if (value in obj) {
+            delete obj[value];
+          }
+        }
+        break;
+
+      default:
```

Comment:
```suggestion
      case 'clear':
```
Краще явно описувати кожен кейс